clippy_lints/src/non_send_fields_in_send_ty.rs

```diff
@@ -118,14 +122,14 @@ impl<'tcx> LateLintPass<'tcx> for NonSendFieldInSendTy {
                         NON_SEND_FIELDS_IN_SEND_TY,
                         item.span,
                         &format!(
-                            "this implementation is unsound, as some fields in `{}` are `!Send`",
+                            "there are some fields in `{}` are not safe to be sent to another thread",
                             snippet(cx, hir_impl.self_ty.span, "Unknown")
                         ),
                         |diag| {
                             for field in non_send_fields {
                                 diag.span_note(
                                     field.def.span,
-                                    &format!("the type of field `{}` is `!Send`", field.def.ident.name),
+                                    &format!("it is not safe to send field `{}` to another thread", field.def.ident.name),
```

Comment:
Here I liked the mentioning of the `Send` trait. What about:
```suggestion
                                    &format!("it is not safe to send field `{}` to another thread, as the type does not implement `Send`", field.def.ident.name),
```

Reply:
Omitting `Send` from the description was in fact intentional. `!Send` has subsets, (1) types that are fundamentally unsafe to send (e.g., `Rc`) and (2) types that are marked as `!Send` ["as a lint"](https://doc.rust-lang.org/nomicon/send-and-sync.html) such as pointers. The goal of this lint is to warn about the first subset and avoid raise warning for the second subset, and it seems that mentioning `Send` trait leads to unintended interpretation of the lint's purpose according to the people's response.

Reply:
Ahh, okay, you're right :+1: